src/django/api/views.py

```diff
@@ -159,6 +164,65 @@ def all_countries(request):
     return Response(COUNTRY_CHOICES)
 
 
+class FacilitiesViewSet(ReadOnlyModelViewSet):
+    queryset = Facility.objects.all()
+    serializer_class = FacilitySerializer
+    permission_classes = (AllowAny,)
+
+    def list(self, request):
+        name = request.query_params.get(FacilitiesQueryParams.NAME,
+                                        None)
+        contributors = request.query_params \
+                              .getlist(FacilitiesQueryParams.CONTRIBUTORS)
+        contributor_types = request \
+            .query_params \
+            .getlist(FacilitiesQueryParams.CONTRIBUTOR_TYPES)
+        countries = request.query_params \
+                           .getlist(FacilitiesQueryParams.COUNTRIES)
+
+        queryset = Facility.objects.all()
+
+        if name is not None:
+            queryset = queryset.filter(name__icontains=name)
+
+        if countries is not None and len(countries):
+            queryset = queryset.filter(country_code__in=countries)
+
+        if len(contributor_types):
+            type_match_facility_ids = [
+                match.facility.id
+                for match
+                in FacilityMatch
```

Comment:
The `FacilityMatch` table may contain rows that are pending or rejected suggestions from the matching algorithm. We will want to add a `filter(status__in[FacilityMatch.AUTOMATIC, FacilityMatch.CONFIRMED])` to exclude pending or rejected matches.